homeassistant/components/rointe/climate.py

```diff
@@ -0,0 +1,272 @@
+        """Init the Climate entity."""
+
+        super().__init__(
+            coordinator, radiator, name=radiator.name, unique_id=radiator.id
+        )
+
+        self.entity_description = ClimateEntityDescription(
+            key="radiator",
+            name=radiator.name,
+        )
+
+    @property
+    def icon(self) -> str | None:
+        """Icon of the entity."""
+        return "mdi:radiator"
+
+    @property
+    def temperature_unit(self) -> str:
+        """Temperature unit."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the current temperature."""
+        if self._radiator.mode == RADIATOR_MODE_MANUAL:
+            if self._radiator.preset == RADIATOR_PRESET_ECO:
+                return self._radiator.eco_temp
+            if self._radiator.preset == RADIATOR_PRESET_COMFORT:
+                return self._radiator.comfort_temp
+            if self._radiator.preset == RADIATOR_PRESET_ICE:
+                return self._radiator.ice_temp
+
+        return self._radiator.temp
+
+    @property
+    def current_temperature(self) -> float:
+        """Get current temperature (Probe)."""
+        return self._radiator.temp_probe
+
+    @property
+    def max_temp(self) -> float:
+        """Max selectable temperature."""
+        if self._radiator.user_mode_supported and self._radiator.user_mode:
+            return self._radiator.um_max_temp
+
+        return RADIATOR_TEMP_MAX
+
+    @property
+    def min_temp(self) -> float:
+        """Minimum selectable temperature."""
+        if self._radiator.user_mode_supported and self._radiator.user_mode:
+            return self._radiator.um_min_temp
+
+        return RADIATOR_TEMP_MIN
+
+    @property
+    def target_temperature_high(self) -> float:
+        """Max selectable target temperature."""
+        if self._radiator.user_mode_supported and self._radiator.user_mode:
+            return self._radiator.um_max_temp
+
+        return RADIATOR_TEMP_MAX
+
+    @property
+    def target_temperature_low(self) -> float:
+        """Minimum selectable target temperature."""
+        if self._radiator.user_mode_supported and self._radiator.user_mode:
+            return self._radiator.um_min_temp
+
+        return RADIATOR_TEMP_MIN
+
+    @property
+    def supported_features(self) -> ClimateEntityFeature:
+        """Flag supported features."""
+        return (
+            ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+        )
+
+    @property
+    def target_temperature_step(self) -> float | None:
+        """Temperature step."""
+        return RADIATOR_TEMP_STEP
+
+    @property
+    def hvac_modes(self) -> list[str]:
+        """Return hvac modes available."""
+        return [HVACMode.OFF, HVACMode.HEAT, HVACMode.AUTO]
+
+    @property
+    def preset_modes(self) -> list[str]:
+        """Return the available preset modes."""
+        return [PRESET_COMFORT, PRESET_ECO, PRESET_ROINTE_ICE]
```

Comment:
I would make this a constant (so that it isn't run fresh every time `preset_modes` is accessed).